sensor_test.py

```diff
@@ -0,0 +1,214 @@
+    sensor_info = get_sensor_info()
+
+    print("3. Validate that sensor reading interval is set to interval from Step 1")
+    assert (
+        sensor_info["reading_interval"] == interval
+    ), "Sensor reading interval was not set correctly"
+
+    print("4. Get sensor reading")
+    sensor_reading = get_sensor_reading()
+
+    print("5. Wait for interval specified in Step 1")
+
+    print("6. Get sensor reading")
+    sensor_reading_after_interval = wait(
+        func=get_sensor_reading(),
+        condition=lambda x: isinstance(x, float),
+        tries=15,
+        timeout=interval,
+    )
+
+    print(
+        f"Validate that reading from Step 4 doesn't equal reading from Step 6, current readings are: {sensor_reading} and {sensor_reading_after_interval}"
+    )
+    assert (
+        sensor_reading != sensor_reading_after_interval
+    ), "Sensor reading interval is not working correctly"
+
+
+# Максимальна версія прошивки сенсора -- 15,
+def test_update_sensor_firmware(get_sensor_info, update_sensor_firmware):
+    """
+    1. Get original sensor firmware version.
+    2. Request firmware update.
+    3. Get current sensor firmware version.
+    4. Validate that current firmware version is +1 to original firmware version.
+    5. Repeat steps 1-4 until sensor is at max_firmware_version - 1.
+    6. Update sensor to max firmware version.
+    7. Validate that sensor is at max firmware version.
+    8. Request another firmware update.
+    9. Validate that sensor doesn't update and responds appropriately.
+    10. Validate that sensor firmware version doesn't change if it's at maximum value.
+    """
+
+    max_firmware_version = 15
+    firmware_version = "firmware_version"
+
+    print("Get original sensor firmware version")
+    sensor_info = get_sensor_info()
+    initial_firmware_version = sensor_info[firmware_version]
+
+    while initial_firmware_version < max_firmware_version:
+        print("Request firmware update")
+        update_sensor_firmware()
+
+        print("Get current sensor firmware version")
+        sensor_info_after_update = wait(
+            func=get_sensor_info,
+            condition=lambda x: isinstance(x, dict),
+            tries=15,
+            timeout=1,
+        )
+        updated_firmware_version = sensor_info_after_update[firmware_version]
+
+        print(
+            "Validate that current firmware version is +1 to original firmware version"
+        )
+        assert (
+            updated_firmware_version == initial_firmware_version + 1
+        ), "Sensor firmware version was not updated correctly"
+
+        initial_firmware_version = updated_firmware_version
+
+    print("Update sensor to max firmware version")
+    print("Validate that sensor is at max firmware version")
+    assert (
+        initial_firmware_version == max_firmware_version
+    ), "Sensor firmware version not max"
+
+    print("Request another firmware update")
+    firmware_message = update_sensor_firmware()
```

Comment:
This variable name needs a bit of work. What you get from `update_sensor_firmware()` isn't really a "firmware_message" but more like `response_to_firmware_update_request`.